File: 1894-merge-strings-alternately/merge-strings-alternately.ts
function mergeAlternately(word1: string, word2: string): string {
    const merged: string[] = []

    const length1: number = word1.length
    const length2: number = word2.length
    // const maxLength: number = length1 > length2 ? length1 : length2

    for (let i = 0; i < Math.max(length1, length2); i++) {
        if (i < length1) {
                    merged.push(word1[i])

        }

        if (i < length2) {
            merged.push(word2[i])
        }
    }

    return merged.join('')
};

File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i = j = 0
        m, n = len(word1), len(word2)
        res = []
        count = 0

        while i < m and j < n:
            if count % 2 == 0:
                res.append(word1[i])
                i += 1
            else:
                res.append(word2[j]) 
                j += 1
            count += 1
        
        res.extend(word1[i:])
        res.extend(word2[j:])

        return "".join(res)